File: mycuration/src/androidTest/java/com/phicdy/mycuration/domain/task/GetFeedIconTaskTest.java
package com.phicdy.mycuration.domain.task;

import android.support.test.runner.AndroidJUnit4;

import com.phicdy.mycuration.util.FileUtil;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.io.File;

import static android.support.test.InstrumentationRegistry.getTargetContext;
import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

@RunWith(AndroidJUnit4.class)
public class GetFeedIconTaskTest {

    @Before
	public void setup() {
    	FileUtil.INSTANCE.setUpIconSaveFolder(getTargetContext());
	}

    @After
    public void tearDown() {
        File file = new File(FileUtil.INSTANCE.iconSaveFolder() + "/kindou.info.png");
        if (file.exists()) file.delete();
        File greeBlogIcon = new File(FileUtil.INSTANCE.iconSaveFolder() + "/labs.gree.jp.png");
        if (greeBlogIcon.exists()) greeBlogIcon.delete();
    }

    @Test
	public void iconExistsWhenGetKindouIcon() {
        File file = new File(FileUtil.INSTANCE.iconSaveFolder() + "/kindou.info.png");
        if (file.exists()) assertTrue(file.delete());
		String iconSaveFolderStr = FileUtil.INSTANCE.iconSaveFolder();
		GetFeedIconTask task = new GetFeedIconTask(iconSaveFolderStr);
		task.execute("http://kindou.info");
		try {
			Thread.sleep(3000);
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
		file = new File(FileUtil.INSTANCE.iconSaveFolder() + "/kindou.info.png");
		assertTrue(file.exists());
	}

	@Test
	public void iconDoesNotExistWhenGetGreeBlogIcon() {
        File greeBlogIcon = new File(FileUtil.INSTANCE.iconSaveFolder() + "/labs.gree.jp.png");
        if (greeBlogIcon.exists()) assertTrue(greeBlogIcon.delete());
		String iconSaveFolderStr = FileUtil.INSTANCE.iconSaveFolder();
		GetFeedIconTask greeBlogIconTask = new GetFeedIconTask(iconSaveFolderStr);
		greeBlogIconTask.execute("http://labs.gree.jp/blog");
		try {
			Thread.sleep(3000);
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
        greeBlogIcon = new File(FileUtil.INSTANCE.iconSaveFolder() + "/labs.gree.jp.png");
		assertFalse(greeBlogIcon.exists());
	}
}

File: mycuration/src/androidTest/java/com/phicdy/mycuration/domain/task/GetFeedIconTaskTest.kt
package com.phicdy.mycuration.domain.task

import android.support.test.InstrumentationRegistry
import android.support.test.runner.AndroidJUnit4
import com.phicdy.mycuration.data.db.DatabaseAdapter
import com.phicdy.mycuration.data.db.DatabaseHelper
import com.phicdy.mycuration.data.repository.ArticleRepository
import com.phicdy.mycuration.data.repository.FilterRepository
import com.phicdy.mycuration.data.repository.RssRepository
import com.phicdy.mycuration.data.rss.Feed
import kotlinx.coroutines.runBlocking
import org.hamcrest.CoreMatchers.`is`
import org.junit.After
import org.junit.Assert.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class GetFeedIconTaskTest {

    private lateinit var rssRepository: RssRepository

    @Before
    fun setup() {
        val db = DatabaseHelper(InstrumentationRegistry.getTargetContext()).writableDatabase
        rssRepository = RssRepository(db, ArticleRepository(db), FilterRepository(db))
    }

    @After
    fun tearDown() {
    }

    @Test
    fun iconExistsWhenGetKindouIcon() = runBlocking {
        val kindou = "http://kindou.info"
        rssRepository.store("kindou", kindou, Feed.ATOM, kindou)
        val task = GetFeedIconTask()
        val iconPath = task.execute(kindou)
        assertThat(iconPath, `is`("https://kindou.info/img/favicon.ico"))
    }

    @Test
    fun iconDoesNotExistWhenGetGreeBlogIcon() = runBlocking {
        val gree = "http://labs.gree.jp/blog"
        rssRepository.store("gree", gree, Feed.ATOM, gree)
        val greeBlogIconTask = GetFeedIconTask()
        val iconPath = greeBlogIconTask.execute(gree)
        assertThat(iconPath, `is`(""))
    }
}
